dmriprep/config/__init__.py

```diff
@@ -1,6 +1,536 @@
+
+_templateflow_home = Path(os.getenv(
+    'TEMPLATEFLOW_HOME',
+    os.path.join(os.getenv('HOME'), '.cache', 'templateflow'))
+)
+
+try:
+    from psutil import virtual_memory
+    _free_mem_at_start = round(virtual_memory().free / 1024**3, 1)
+except Exception:
+    _free_mem_at_start = None
+
+_oc_limit = 'n/a'
+_oc_policy = 'n/a'
+try:
+    # Memory policy may have a large effect on types of errors experienced
+    _proc_oc_path = Path('/proc/sys/vm/overcommit_memory')
+    if _proc_oc_path.exists():
+        _oc_policy = {
+            '0': 'heuristic', '1': 'always', '2': 'never'
+        }.get(_proc_oc_path.read_text().strip(), 'unknown')
+        if _oc_policy != 'never':
+            _proc_oc_kbytes = Path('/proc/sys/vm/overcommit_kbytes')
+            if _proc_oc_kbytes.exists():
+                _oc_limit = _proc_oc_kbytes.read_text().strip()
+            if _oc_limit in ('0', 'n/a') and Path('/proc/sys/vm/overcommit_ratio').exists():
+                _oc_limit = '{}%'.format(
+                    Path('/proc/sys/vm/overcommit_ratio').read_text().strip()
+                )
+except Exception:
+    pass
+
+
+class _Config:
+    """An abstract class forbidding instantiation."""
+
+    _paths = tuple()
+
+    def __init__(self):
+        """Avert instantiation."""
+        raise RuntimeError('Configuration type is not instantiable.')
+
+    @classmethod
+    def load(cls, settings, init=True):
+        """Store settings from a dictionary."""
+        for k, v in settings.items():
+            if v is None:
+                continue
+            if k in cls._paths:
+                setattr(cls, k, Path(v).absolute())
+                continue
+            if hasattr(cls, k):
+                setattr(cls, k, v)
+
+        if init:
+            try:
+                cls.init()
+            except AttributeError:
+                pass
+
+    @classmethod
+    def get(cls):
+        """Return defined settings."""
+        out = {}
+        for k, v in cls.__dict__.items():
+            if k.startswith('_') or v is None:
+                continue
+            if callable(getattr(cls, k)):
+                continue
+            if k in cls._paths:
+                v = str(v)
+            if isinstance(v, _SRs):
+                v = ' '.join([str(s) for s in v.references]) or None
+            if isinstance(v, _Ref):
+                v = str(v) or None
+            out[k] = v
+        return out
+
+
+class environment(_Config):
+    """
+    Read-only options regarding the platform and environment.
+
+    Crawls runtime descriptive settings (e.g., default FreeSurfer license,
+    execution environment, nipype and *dMRIPrep* versions, etc.).
+    The ``environment`` section is not loaded in from file,
+    only written out when settings are exported.
+    This config section is useful when reporting issues,
+    and these variables are tracked whenever the user does not
+    opt-out using the ``--notrack`` argument.
```

Comment:
The latter is not true with popylar

Reply:
What do you mean? That people can't drop out or that these variables are not being tracked?

If the user does not opt-out, do you think it'd be possible to add these variables as metadata somehow?

Reply:
Sorry for the ambiguity: what I meant is that these variables are not recorded. It would be tricky to add them, as  the payload for popylar is rather limited (see https://github.com/popylar/popylar/blob/master/popylar/popylar.py#L106-L115)